src/quacc/utils/dicts.py

```diff
@@ -63,13 +64,16 @@ def _recursive_dict_pair_merge(
     dict
         Merged dictionary
     """
-    dict1 = dict1 or {}
-    dict2 = dict2 or {}
+
+    dict1 = dict1 or (dict1.__class__() if dict1 is not None else {})
+    dict2 = dict2 or (dict2.__class__() if dict2 is not None else {})
```

Comment:
Could you walk me through this one? The only time the first expression in the `or` boolean would be `False` is if `dict1` is `None` or `False` or some other falsiness value (e.g. `[]`, `{}`, 0). How/when does it get to the second expression `(dict1.__class__() if dict1 is not None else {})`?

Reply:
The plan is to make sure if you send an empty kind of dict class it does not come back as a 'normal dict' which was the case before?

Reply:
Oh, I only just learned that `{}` has a `.__class__()` attribute! I never even thought about the attributes of `{}`.

Reply:
The biggest surprise for me was that None has a .__class__ attribute...

I guess it does not really matter, it is just to avoid any surprise to the user who would use this function and be surprised by the class change.